packages/block-editor/src/hooks/utils.js

```diff
@@ -16,22 +16,28 @@ import { useSetting } from '../components';
 import { useSettingsForBlockElement } from '../components/global-styles/hooks';
 
 /**
- * Removed falsy values from nested object.
+ * Removed falsy values from nested object. If `allowZeros` is `true`, then
+ * zero values will be allowed within the resulting object.
+ *
+ * @param {*}       object             Object to be cleaned.
+ * @param {Object}  options            Options config.
+ * @param {boolean} options.allowZeros Whether to allow zero values.
  *
- * @param {*} object
  * @return {*} Object cleaned from falsy values
  */
-export const cleanEmptyObject = ( object ) => {
+export const cleanEmptyObject = ( object, { allowZeros } = {} ) => {
 	if (
 		object === null ||
 		typeof object !== 'object' ||
 		Array.isArray( object )
 	) {
 		return object;
 	}
+	const recursiveCleanObject = ( nestedObject ) =>
+		cleanEmptyObject( nestedObject, { allowZeros } );
 	const cleanedNestedObjects = Object.fromEntries(
-		Object.entries( mapValues( object, cleanEmptyObject ) ).filter(
-			( [ , value ] ) => Boolean( value )
+		Object.entries( mapValues( object, recursiveCleanObject ) ).filter(
+			( [ , value ] ) => Boolean( value ) || ( allowZeros && value === 0 )
```

Comment:
It is getting a little unreadable, so I think it might be good to break it onto multiple lines:

```suggestion
			( [ , value ] ) => {
			    if ( options?.allowZeros && value === 0 ) return true;
			    return Boolean( value );
			}
```